Geometry/HGCalCommonData/src/HGCalGeomParameters.cc

```diff
@@ -654,16 +906,149 @@ void HGCalGeomParameters::loadCellParsHexagon(const DDCompactView* cpv,
   }
 
 #ifdef EDM_ML_DEBUG
-  std::cout << "HGCalLoadCellPars: " << php.cellFine_.size() 
-	    << " rows for fine cells\n";
+  edm::LogVerbatim("HGCalGeom") << "HGCalLoadCellPars: " 
+				<< php.cellFine_.size()
+				<< " rows for fine cells";
   for (unsigned int k=0; k<php.cellFine_.size(); ++k)
-    std::cout << k << ":" << php.cellFine_[k] << " ";
-  std::cout << std::endl;
-  std::cout << "HGCalLoadCellPars: " <<php.cellCoarse_.size()
-            << " rows for coarse cells\n";
+    edm::LogVerbatim("HGCalGeom") << "[" << k << "]: " << php.cellFine_[k];
+  edm::LogVerbatim("HGCalGeom") << "HGCalLoadCellPars: " 
+				<< php.cellCoarse_.size()
+				<< " rows for coarse cells";
   for (unsigned int k=0; k<php.cellCoarse_.size(); ++k)
-    std::cout << k << ":" << php.cellCoarse_[k] << " ";
-  std::cout << std::endl;
+    edm::LogVerbatim("HGCalGeom") << "[" << k << "]: " << php.cellCoarse_[k];
+#endif
+}
+
+void HGCalGeomParameters::loadSpecParsHexagon8(const DDFilteredView& fv,
+					       HGCalParameters& php) {
+
+  DDsvalues_type sv(fv.mergedSpecifics());
+  php.cellThickness_ = DDVectorGetter::get("CellThickness");
+  for (double & k : php.cellThickness_) 
+    k *= k_ScaleFromDDD;
```

Comment:
@bsunanda - this code fragment calls for use of a lambda expression:
std::for_each(php.cellThickness_.begin(), php.cellThickness_.end(), [](int &n){ n*=k_ScaleFromDDD; });